### DISCUSSION TITLE
Coverage with Workspace

### DISCUSSION THREAD

**Discussion by zirkelc** [vitest-dev/vitest#3852] [Q&A] [answered]
I have a Monorepo with the two packages and use Vitest Workspaces to define the projects:

**vitest.workspace.js**
```js
import { defineWorkspace } from 'vitest/config'

export default defineWorkspace(['packages/*']);
```

**/packages/(foo|bar)/vitest.config.js**
```js
import { defineConfig } from 'vitest/config'

export default defineConfig({
	test: {
		coverage: {
			all: true,
			reporter: ['text', 'json-summary', 'json'],
		},
	},
})
```

When I run Vitest from the monorepo root with coverage enabled, the actual settings `coverage` settings from the individual `vitest.config.js` files are ignored. That's stated in the [docs](https://vitest.dev/guide/workspace.html#configuration).

However, I was wondering how to configure the coverage options for the whole workspace? 
I currently supply the coverage options as CLI flags:
```sh
vitest --coverage.enabled --coverage.provider=v8 --coverage.reporter=json-summary --coverage.reporter=json
```

Is there a way to set workspace defaults in a config file?

**Comment by AriPerkkio** [comment, marked as answer]
> However, I was wondering how to configure the coverage options for the whole workspace?
> Is there a way to set workspace defaults in a config file?

Create a `vitest.config.ts` in the workspace root, right next to the `vitest.workspace.ts`. Add coverage related configuration there. Exactly as done here: https://github.com/vitest-dev/vitest/tree/main/test/workspaces.

**Reply by zirkelc** [reply]
thank you! I've tried that already, but made a mistake as I added `singleThread: true` to this workspace config as well. That obviously altered the entire test configuration and it made me assume it's not possible to have a config in the root when using workspaces.
